cmd/restic/cmd_backup.go

```diff
@@ -175,12 +175,12 @@ func readLinesFromFile(filename string) ([]string, error) {
 
 // Check returns an error when an invalid combination of options was set.
 func (opts BackupOptions) Check(gopts GlobalOptions, args []string) error {
-	if opts.FilesFrom == "-" && gopts.password == "" {
+	if opts.FilesFrom != nil && opts.FilesFrom[0] == "-" && gopts.password == "" {
 		return errors.Fatal("unable to read password from stdin when data is to be read from stdin, use --password-file or $RESTIC_PASSWORD")
 	}
 
 	if opts.Stdin {
-		if opts.FilesFrom != "" {
+		if opts.FilesFrom != nil {
```

Comment:
A slice can be non-nil and still have zero elements, better check its length:
```suggestion
		if len(opts.FilesFrom) > 0 {
```